packages/cpp-debug/src/browser/wrapper-widgets/memory-layout-widget.tsx

```diff
@@ -0,0 +1,162 @@
+    static readonly ID = 'memory-layout-widget';
+    static readonly LABEL = 'Memory Inspector';
+
+    // Necessary to inherit theia's tabbar styling
+    static readonly DOCK_PANEL_ID = 'theia-main-content-panel';
+    static readonly THEIA_TABBAR_CLASSES = ['theia-app-centers', 'theia-app-main'];
+    static readonly CPP_TABBAR_CLASS = 'memory-dock-tabbar';
+    static readonly DOCK_PANEL_CLASS = 'memory-dock-panel';
+
+    @inject(WidgetManager) protected readonly widgetManager: WidgetManager;
+    @inject(MemoryWidgetManager) protected readonly memoryWidgetManager: MemoryWidgetManager;
+    @inject(MemoryDiffSelectWidget) protected readonly diffSelectWidget: MemoryDiffSelectWidget;
+    @inject(MemoryDockpanelPlaceholder) protected readonly placeholderWidget: MemoryDockpanelPlaceholder;
+    @inject(ApplicationShell) protected readonly shell: ApplicationShell;
+
+    protected readonly onDidChangeTrackableWidgetsEmitter = new Emitter<Widget[]>();
+    readonly onDidChangeTrackableWidgets = this.onDidChangeTrackableWidgetsEmitter.event;
+
+    protected readonly toDispose = new DisposableCollection();
+    protected dockPanel: MemoryDockPanel;
+    protected hasGeneratedWidgetAutomatically = false;
+
+    @postConstruct()
+    protected async init(): Promise<void> {
+        this.id = MemoryLayoutWidget.ID;
+        this.addClass(MemoryLayoutWidget.ID);
+        this.title.label = MemoryLayoutWidget.LABEL;
+        this.title.caption = MemoryLayoutWidget.LABEL;
+        this.title.closable = true;
+        this.title.iconClass = 'memory-view-icon';
+        this.dockPanel = await this.widgetManager.getOrCreateWidget<MemoryDockPanel>(MemoryDockPanel.ID);
+        this.addWidget(this.dockPanel);
+        this.addWidget(this.diffSelectWidget);
+        this.addWidget(this.placeholderWidget);
+        this.toDispose.push(this.memoryWidgetManager.onDidCreateNewWidget(widget => {
+            this.dockPanel.addWidget(widget);
+            this.dockPanel.activateWidget(widget);
+            this.onDidChangeTrackableWidgetsEmitter.fire([widget]);
+        }));
+        this.toDispose.push(this.memoryWidgetManager.onChanged(() => {
+            if (!this.memoryWidgetManager.canCompare) {
+                this.diffSelectWidget.hide();
+            }
+        }));
+        this.dockPanel.widgetRemoved.connect(this.handleWidgetRemoved.bind(this), this);
+        this.dockPanel.widgetAdded.connect(this.handleWidgetsChanged.bind(this), this);
+        this.toDispose.push(this.onDidChangeTrackableWidgetsEmitter);
+        this.diffSelectWidget.hide();
+        this.update();
+    }
+
+    toggleComparisonVisibility(): void {
+        if (this.diffSelectWidget.isHidden) {
+            this.diffSelectWidget.show();
+        } else {
+            this.diffSelectWidget.hide();
+        }
+        this.update();
+    }
+
+    dispose(): void {
+        this.toDispose.dispose();
+    }
```

Comment:
It looks like the `super` call is missing here and it's not possible to properly dispose of this widget. Can we add this?
```suggestion
        this.super.dispose();
    }
```